src/vellum/workflows/workflows/base.py

```diff
@@ -611,12 +612,80 @@ def validate(self) -> None:
         # https://app.shortcut.com/vellum/story/4327
         pass
 
+    def _resolve_node_ref(self, node_ref: Union[Type[BaseNode], UUID, str]) -> Type[BaseNode]:
+        """
+        Resolve a node reference (class, UUID, or string) to a node class.
+
+        Args:
+            node_ref: Either a node class, a UUID, or a fully qualified string
+                     in the format "module.ClassName"
+
+        Returns:
+            The resolved node class
+
+        Raises:
+            ValueError: If the node reference cannot be resolved
+        """
+        if inspect.isclass(node_ref) and issubclass(node_ref, BaseNode):
+            return node_ref
+
+        candidate_nodes: List[Type[BaseNode]] = []
+        for node in self.get_nodes():
+            candidate_nodes.append(node)
+            wrapped_node = get_wrapped_node(node)
+            while wrapped_node:
+                candidate_nodes.append(wrapped_node)
+                wrapped_node = get_wrapped_node(wrapped_node)
+
+        if isinstance(node_ref, UUID):
+            for node in candidate_nodes:
+                if node.__id__ == node_ref:
+                    return node
+            raise ValueError(
+                f"No node found with UUID {node_ref}. "
+                f"Available nodes: {[f'{n.__name__} ({n.__id__})' for n in candidate_nodes]}"
```

Comment:
Simplify this error message to just `f"Node '{identifier}' not found in workflow"` to match what we do in our workflow server